### PR TITLE
Migrate off deprecated `AwsS3V4Signer` to `AwsV4HttpSigner`

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Signer.java [hunk 1]
```diff
@@ -66,43 +64,38 @@ public SigningResponse sign(SigningRequest clientRequest) {
...
+
+    SignRequest.Builder<AwsCredentialsIdentity> signRequest =
+        SignRequest.builder(credentials)
+            .request(request.build())
+            .putProperty(AwsV4HttpSigner.REGION_NAME, clientRequest.region())
+            .putProperty(AwsV4HttpSigner.SERVICE_SIGNING_NAME, S3Client.SERVICE_NAME)
+            .putProperty(AwsV4HttpSigner.DOUBLE_URL_ENCODE, false)
+            .putProperty(AwsV4HttpSigner.NORMALIZE_PATH, false)
+            .putProperty(AwsV4HttpSigner.CHUNK_ENCODING_ENABLED, false)
+            .putProperty(AwsV4HttpSigner.PAYLOAD_SIGNING_ENABLED, false);
```
Comment (adutra) [inline catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Signer.java:88]: Should we enforce the algorithm?
```suggestion
            .putProperty(AwsV4HttpSigner.PAYLOAD_SIGNING_ENABLED, false)
            .putProperty(AwsV4HttpSigner.CHECKSUM_ALGORITHM, DefaultChecksumAlgorithm.SHA256);
```

Comment (snazy) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Signer.java:88]: That's been there before. 🤷

Comment (snazy) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Signer.java:88]: And the suggested change actually breaks the functionality :(

**Review by adutra — APPROVED** [review]

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Signer.java (hunk 1 above)
Comment (snazy) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Signer.java:88]: BTW: It uses SHA256 (I looked into the headers)

**Pull request merged**